pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -186,8 +216,58 @@ func (s *AWSCloud) ensureLoadBalancer(namespacedName types.NamespacedName, loadB
 				if err != nil {
 					return nil, fmt.Errorf("error creating AWS loadbalancer listeners: %v", err)
 				}
+
+				if proxyProtocol {
+					for _, newListener := range additions {
+						// TODO(williamsandrew) DRY this up
```

Comment:
Agreed - I think it's probably a matter of doing a similar reconciliation block to the one we're doing here, except you just check whether the policies match.  It looks like you do have the policies on ListenerDescription.  It should also be a little bit easier, because there's no need to combine requests, because the API doesn't give that ability anyway.